lib/cisco_node_utils/snmpserver.rb

```diff
@@ -106,16 +106,14 @@ def default_global_enforce_priv
     end
 
     def protocol?
-      match = config_get('snmp_server', 'protocol')
-      !match.nil? && match.include?('Enable')
+      config_get('snmp_server', 'protocol')
     end
 
     def protocol=(enable)
-      if enable
-        config_set('snmp_server', 'protocol', '')
-      else
-        config_set('snmp_server', 'protocol', 'no')
-      end
+      no_cmd = (enable ? '' : 'no')
+      config_set('snmp_server', 'protocol', no_cmd)
+    rescue Cisco::CliError => e
```

Comment:
You should only rescue a CliError if you need to catch a specific failure case (for example, we know that some commands may raise a CliError if a specific 'no feature' command is configured). Otherwise, yes, please remove this rescue logic.